### PR TITLE
Expense creation from Slack (both form as well as DE flow)

### PR DESCRIPTION
**Pull request by shreyanshss7**

### CODE REVIEW

File: fyle_slack_app/slack/interactives/block_action_handlers.py [hunk 1]
```diff
@@ -141,6 +155,314 @@ def handle_notification_preference_selection(self, slack_payload: Dict, user_id:
...
+
+        view_id = slack_payload['container']['view_id']
+
+        slack_client = get_slack_client(team_id)
+
+        expense_id = slack_payload['view']['private_metadata']
+
+        fyle_expense = FyleExpense(user)
+
+        expense_id = 'txCCVGvNpDMM'
```
Comment (shreyanshss7) [inline fyle_slack_app/slack/interactives/block_action_handlers.py:349]: Pls ignore hard coded ids

File: fyle_slack_app/fyle/expenses/views.py [hunk 2]
```diff
@@ -1,18 +1,252 @@
...
+            'fields_render_property': fields_render_property,
+            'selected_project': project,
+            'additional_currency_details': additional_currency_details,
+            'add_to_report': add_to_report,
+            'custom_fields': custom_field_blocks
+        }
+        return current_form_details
+
+
     def get_expense_by_id(self, expense_id: str) -> Dict:
```
Comment (jatzz10) [inline fyle_slack_app/fyle/expenses/views.py:250]: @shreyanshs7 Can you please move this method above the static methods?

File: fyle_slack_app/slack/interactives/block_action_handlers.py (hunk 1 above)
Comment (jatzz10) [reply fyle_slack_app/slack/interactives/block_action_handlers.py:445]: @shreyanshs7 please remove hard coded stuff

File: fyle_slack_app/slack/interactives/block_action_handlers.py [hunk 3]
```diff
@@ -209,6 +428,132 @@ def handle_feedback_dialog(self, slack_payload: Dict, user_id: str, team_id: str
...
+
+        user = utils.get_or_none(User, slack_user_id=user_id)
+
+        expense_id = slack_payload['actions'][0]['value']
+
+        trigger_id = slack_payload['trigger_id']
+
+        slack_client = get_slack_client(team_id)
+
+        expense_id = 'txCCVGvNpDMM'
```
Comment (jatzz10) [inline fyle_slack_app/slack/interactives/block_action_handlers.py:473]: Here as well

File: fyle_slack_app/slack/interactives/block_action_handlers.py [hunk 4]
```diff
@@ -209,6 +428,132 @@ def handle_feedback_dialog(self, slack_payload: Dict, user_id: str, team_id: str
...
+        async_task(
+            'fyle_slack_app.slack.interactives.tasks.handle_submit_report_dialog',
+            user,
+            team_id,
+            report_id,
+            response['view']['id']
+        )
+
+        return JsonResponse({})
+
```
Comment (jatzz10) [inline fyle_slack_app/slack/interactives/block_action_handlers.py:556]: can add one more line space

File: fyle_slack_app/slack/interactives/tasks.py [hunk 5]
```diff
@@ -1,20 +1,267 @@
...
+    # Removing loading info from below project input element
+    project_loading_block_index = next((index for (index, d) in enumerate(current_ui_blocks) if d['block_id'] == 'project_loading_block'), None)
+    current_ui_blocks.pop(project_loading_block_index)
+
+    form_metadata['project'] = project
+
+    cache.set(cache_key, form_metadata)
+
+    new_expense_dialog_form = expense_dialog_form(
+        **current_expense_form_details
```
Comment (jatzz10) [inline fyle_slack_app/slack/interactives/tasks.py:67]: `**` ?

File: fyle_slack_app/slack/interactives/view_submission_handlers.py [hunk 6]
```diff
@@ -52,10 +64,89 @@ def handle_view_submission(self, slack_payload: Dict, user_id: str, team_id: str
...
+
+        report_submitted_message = expense_messages.report_submitted_message(user, report['data'][0])
+
+        slack_client.chat_postMessage(channel=user.slack_dm_channel_id, blocks=report_submitted_message)
+
+        return JsonResponse({})
+
+
+    def handle_add_expense_to_report(self, slack_payload: Dict, user_id: str, team_id: str) -> JsonResponse:
+
```
Comment (jatzz10) [inline fyle_slack_app/slack/interactives/view_submission_handlers.py:131]: can remove extra line space

**Review by jatzz10 — APPROVED** [review]
@shreyanshs7 added few minor comments - can you those once?

**Pull request merged**